src/AppInstallerCLICore/Commands/CompleteCommand.cpp

```diff
@@ -53,8 +53,10 @@ namespace AppInstaller::CLI
             }
 
             // Create a new Context to execute the Complete from
-            auto subContextPtr = context.Clone();
+            auto subContextPtr = context.CreateSubContext();
             Context& subContext = *subContextPtr;
+            auto previousThreadGlobals = subContext.GetThreadGlobals().SetForCurrentThread();

```

Comment:
```suggestion
            auto contextActive = subContext.SetActive();
```
As well as creating the appropriate `SetActive`.  I think it is easier to follow.